lib/dynflow/delayed_executors/abstract_core.rb

```diff
@@ -46,8 +46,12 @@ def with_error_handling(error_retval = nil, &block)
       def process(delayed_plans, check_time)
         processed_plan_uuids = []
         delayed_plans.each do |plan|
+          fix_plan_state(plan)
```

Comment:
Shouldn't we do this conditionally instead of always calling the method and doing the check there?

Reply:
The reason I did it this way was so that the condition, as well as the fix is on the same place: otherwise, the logic would be split into two parts, which is something I would like to avoid

Reply:
Ok, fair enought